tests/jsm_test.ts

```diff
@@ -1449,3 +1445,55 @@ Deno.test("jsm - consumers with name and durable_name", async () => {
 
   await cleanup(ns, nc);
 });
+
+Deno.test("jsm - consumer name is validated", async () => {
+  const { ns, nc } = await setup(
+    jetstreamServerConf({}, true),
+  );
+
+  if (await notCompatible(ns, nc, "2.9.0")) {
+    return;
+  }
+  const { stream } = await initStream(nc);
+  const jsm = await nc.jetstreamManager();
+
+  function test(
+    n: string,
+    conf: Partial<ConsumerConfig> = {},
+  ): Promise<unknown> {
+    const opts = Object.assign({ name: n, ack_policy: AckPolicy.None }, conf);
+    return jsm.consumers.add(stream, opts);
+  }
+
+  await assertRejects(
+    async () => {
+      await test("hello.world");
+    },
+    Error,
+    "consumer 'name' cannot contain '.'",
+  );
+
+  await assertRejects(
+    async () => {
+      await test("hello>world");
```

Comment:
Technically this would be treated in the server as a literal, so would be a valid subject, but in case some part of the server code does not, it is safer to disallow I guess.

Reply:
these will not only be subjects but files too - and now the issue comes if some file is referenced on a shell, etc.